libraries/AP_AIS/LogStructure.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <AP_Logger/LogStructure.h>
+
+#define LOG_IDS_FROM_AIS \
+    LOG_AIS_RAW_MSG,\
+    LOG_AIS_MSG1, \
+    LOG_AIS_MSG5
+
+struct PACKED log_AIS_raw {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t num;
+    uint8_t total;
+    uint8_t ID;
+    char payload[64];
+};
+
+struct PACKED log_AIS_msg1 {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t type;
+    uint8_t repeat;
+    uint32_t mmsi;
+    uint8_t nav;
+    int8_t rot;
+    uint16_t sog;
+    int8_t pos_acc;
+    int32_t lon;
+    int32_t lat;
+    uint16_t cog;
+    uint16_t head;
+    uint8_t sec_utc;
+    uint8_t maneuver;
+    uint8_t raim;
+    uint32_t radio;
+};
+
+struct PACKED log_AIS_msg5 {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t repeat;
+    uint32_t mmsi;
+    uint8_t ver;
+    uint32_t imo;
+    char call_sign[16];
+    char name[64];
+    uint8_t vessel_type;
+    uint16_t bow_dim;
+    uint16_t stern_dim;
+    uint8_t port_dim;
+    uint8_t star_dim;
+    uint8_t fix;
+    uint8_t draught;
+    char dest[64];
+    uint8_t dte;
+};
+
+// @LoggerMessage: AISR
```

Comment:
in my testing I've only ever seen the AISR messages.  Never the AIS1 or AIS5 messages.. maybe that's just 'cuz I haven't sent many messages through yet.

Reply:
There off by default, I will turn them on too.

Reply:
done